test/test_sassc.rb

```diff
@@ -0,0 +1,364 @@
+require 'sprockets_test'
+
+silence_warnings do
+  require 'sassc'
+end
+
+require 'sprockets/sassc_processor'
+require 'sprockets/sassc_compressor'
+
+class TestBaseSassc < Sprockets::TestCase
+  CACHE_PATH = File.expand_path("../../.sass-cache", __FILE__)
+  COMPASS_PATH = File.join(FIXTURE_ROOT, 'compass')
+
+  def teardown
+    refute ::SassC::Script::Functions.instance_methods.include?(:asset_path)
+    FileUtils.rm_r(CACHE_PATH) if File.exist?(CACHE_PATH)
+    assert !File.exist?(CACHE_PATH)
+  end
+end
+
+class TestNoSasscFunction < TestBaseSass
+  module ::SassC::Script::Functions
+    def javascript_path(path)
+      ::SassC::Script::String.new("/js/#{path.value}", :string)
+    end
+
+    module Compass
+      def stylesheet_path(path)
+        ::SassC::Script::String.new("/css/#{path.value}", :string)
+      end
+    end
+    include Compass
+  end
+
+  test "aren't included globally" do
+    silence_warnings do
+      assert ::SassC::Script::Functions.instance_methods.include?(:javascript_path)
+      assert ::SassC::Script::Functions.instance_methods.include?(:stylesheet_path)
+
+      filename = fixture_path('sass/paths.scss')
+      assert data = File.read(filename)
+      engine = ::SassC::Engine.new(data, {
+        filename: filename,
+        syntax: :scss
+      })
+
+      assert ::SassC::Script::Functions.instance_methods.include?(:javascript_path)
+      assert ::SassC::Script::Functions.instance_methods.include?(:stylesheet_path)
+
+      assert_equal <<-EOS, engine.render
```

Comment:
Slight pro-tip:

You can use `SCSS` instead of `EOS` and some editors like Sublime will change syntax highlighting accordingly for that heredoc scope.